docs/source/gateway.md

```diff
@@ -0,0 +1,118 @@
+
+## How the gateway endorses your transaction proposal
+
+In order for a transaction to be successfully committed to the ledger, a sufficient number of endorsements are required in order to satisfy
+the [endorsement policy](endorsement-policies.html).  Getting an endorsement from an organization involves connecting to one 
+of its peers and have it simulate (execute) the transaction proposal against its copy of the ledger.  The peer does this by
+invoking the chaincode function whose name and arguments are specified in the proposal and building (and signing) a read-write set
+containing the current ledger state and changes it would make in response to the state get/set instructions in that function.
+
+The endorsement policy that gets applied to a transaction depends on the implementation of the chaincode function that 
+is being invoked, and could be a combination of the following:
+
+- **Chaincode policies**.  These are the policies agreed to by channel members when they approve a chaincode definition for their organization.
+If the chaincode function invokes a function in another chaincode, then both policies will need to be satisfied.
+- **Private data collection policies**.  If the chaincode function writes to a state in a private data collection,
+then the endorsement policy for that collection will override the chaincode policy for that state.  If it reads from a private
+collection, then it will be restricted to organizations that own a copy of that collection.
+- **State-based endorsement (SBE) policies**. Also known as key-level signature policies, these can be applied to individual
+states and will override the chaincode policy (or collection policy if it's a private collection state).
+The policies themselves are stored in the ledger and can be modified by transactions.
+
+Although it might seem obvious from the transaction proposal which policies would be applied, the actual combination of
+polices depends on what the chaincode function does at runtime and cannot necessarily be derived from static analysis.
+
+The gateway handles all of this complexity on behalf of the client using the following process:
+
+- The gateway selects the endorsing peer in the gateway's organization (MSPID) that has the highest ledger block height.
+The assumption is that all peers within this organization are *trusted* by the client application that connects
+to this gateway.
+- It simulates the transaction proposal on that peer.  During simulation, it captures all the relevant information on which
+states are accessed and hence which policies will need to be combined (including any individual state-based policies 
+as stored in the peer's copy of the ledger).  
+- This information is assembled into a `ChaincodeInterest` protobuf structure and passed to the discovery service in order
+to derive an endorsement plan specific for this transaction.
```

Comment:
I think we should mention the dependency on discovery service prior to this point. Provide a link to discovery service and mention this is how gateway becomes aware of available ordering service nodes and peers.